### DISCUSSION TITLE
How to reset value of uncontrolled form components?

### DISCUSSION THREAD

**Discussion by Hima11** [react-hook-form/react-hook-form#2545] [Q&A] [answered]
Hi all,

I am using ckEditor in my form for richtext feature. I am trying to clear it's value using reset api from useForm hook. Is this possible with ckEditor and how can we do that?

` <Controller
                        name="IssueDescription"
                        render={({ onChange }) => {
                            return (
                                <CKEditor
                                    editor={ClassicEditor}
                                    onChange={(value: any, editor: { getData: () => any; }) => {
                                        onChange(editor.getData());
                                    }}
                                />
                            )
                        }}
                        control={control}
                    />`

`const resetInitialState = () => {
        reset();
        setDialogMessage("");
        setIsDialogShow(false);
    }`

**Comment by bluebill1049** [comment, marked as answer]
rest({
  ...values,
})

**Reply by Hima11** [reply]
@bluebill1049 sorry I did not get this. Where am I supposed to put this in the code?

**Reply by bluebill1049** [reply]
do you have a codesandbox?

**Reply by Hima11** [reply]
No I do not have as this is a private company project.  I even tried this but did not work **reset(defaultValues);**.  Below is the template form which I am working on.


`import React, { FC, useState } from 'react';
import { Button, Form, FormGroup, FormText, Input, Label } from "reactstrap";
import { useForm, Controller } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers";
import { SharepointSchema } from "../../utils/FormValidationSchema";
// @ts-ignore
import CKEditor from '@ckeditor/ckeditor5-react';
// @ts-ignore
import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
import TemplateSubmitButton from '../../commons/TemplateSubmitButton';
import { Asterisk, ErrorText } from '../../commons/FormUtils';
import useAppContext from '../../store';
import { SubmitRequest } from '.';
import _ from 'lodash';
import RequestSubmitModal from '../../commons/RequestSubmitModal';

const Sharepoint = () => {
    const defaultValues = {
        IssueDescription: ""
    }
    const { store: { selectedBusinessArea } } = useAppContext();
    const [dialogMessage, setDialogMessage] = useState<string>("");
    const [isDialogShow, setIsDialogShow] = useState<boolean>(false);
    const { register, handleSubmit, errors, control, reset } = useForm({
        mode: 'onSubmit',
        reValidateMode: 'onChange',
        resolver: yupResolver(SharepointSchema),
        defaultValues,
        criteriaMode: "all",
        shouldFocusError: true,
        shouldUnregister: true
    });

    const onSubmit = (data: any) => {
        const requestTypeMetadata = {
            RequestTypeId: selectedBusinessArea.RequestType,
            SubCategoryIds: selectedBusinessArea.Name,
            BusinessAreaId: selectedBusinessArea.Id
        };

        SubmitRequest(requestTypeMetadata, data).then((message) => {
            if (!_.isEmpty(message)) {
                setDialogMessage(message);
                setIsDialogShow(true);
            }
        });
    };

    const resetInitialState = () => {
        reset(defaultValues);
        setDialogMessage("");
        setIsDialogShow(false);
    }

    return (
        <>
            <Form onSubmit={handleSubmit(onSubmit)}>
                <FormGroup tag="fieldset">
                    <Label for="url" className="font-weight-bold">What's the URL?: <Asterisk /></Label>
                    <Input
                        innerRef={register}
                        type="text"
                        name="IssueUrl"
                        id="IssueUrl"
                        placeholder="Enter the URL of the page"
                    />
                    <ErrorText errors={errors} name="IssueUrl" />
                </FormGroup>
                <FormGroup>
                    <Label for="description" className="font-weight-bold">How can we make it better?: <Asterisk /></Label>
                    <Controller
                        name="IssueDescription"
                        render={({ onChange }) => {
                            return (
                                <CKEditor
                                    editor={ClassicEditor}
                                    onChange={(value: any, editor: { getData: () => any; }) => {
                                        onChange(editor.getData());
                                    }}
                                />
                            )
                        }}
                        control={control}
                    />
                    <ErrorText errors={errors} name="IssueDescription" />
                </FormGroup>

                <TemplateSubmitButton />
            </Form>
            <RequestSubmitModal header="Request" message={dialogMessage} isShow={isDialogShow} onDialogConfirmed={resetInitialState}></RequestSubmitModal>
        </>
    )
};
export default Sharepoint;`

**Reply by bluebill1049** [reply]
```
<Controller
                    name="IssueDescription"
                    render={({ onChange }) => {
                        return (
                            <CKEditor
                                editor={ClassicEditor}
                                onChange={(value: any, editor: { getData: () => any; }) => {
                                    onChange(editor.getData());
                                }}
                            />
                        )
                    }}
                    control={control}
                />```

you are missing the value prop.

**Reply by Hima11** [reply]
@bluebill1049 Awesome! It worked after adding value. Thanks

`` <Controller
                        name="SharePointIssueDescription"
                        render={({ onChange, value }) => {
                            return (
                                <CKEditor
                                    editor={ClassicEditor}
                                    onChange={(event: any, editor: { getData: () => any; }) => {
                                        onChange(editor.getData());
                                    }}
                                    data={value}
                                />
                            )
                        }}
                        control={control}
                    />```
